### DISCUSSION TITLE
Not recognising zero tariff

### DISCUSSION THREAD

**Discussion by oymyakon** [BottlecapDave/HomeAssistant-OctopusEnergy#251] [General]
Hi,
Firstly i would like to thank dave for a wonderful job on this software.

The problem i am having is that last night we had 3 periods of zero tariff and my target sensors didn't choose them. Is there something i could have set wrong?
Paul.

**Comment by BottlecapDave** [comment]
Hi there and I'm glad this is of use to some people :)

Target rate sensors can not pick the times we expect for a number of reasons

1) The evaluation period - This by default is from midnight to midnight. If a zero tariff started at 11:30pm, then that 30 minute block would not be included as it would count towards the previous days evaluation period. If you're on an agile tariff, then you should probably set these to 16:00 to 16:00 so that it's inline with when the rates are released.
2) The number of hours - This will always pick the periods where the combined cost is the lowest. It _could_ have been that the periods surrounding the zero tariffs were so high that other periods when combined would cost you less. A way around this is to monitor the current price and have "overrides" to turn things on when its zero or below.
3) The cached rates - If your instance had issues downloading the latest prices or the zero tariffs appeared very quickly, the zero tariffs might not have been present in the evaluation.

If you believe these were not the case, please raise an issue with details on
* You tariff
* The target rate sensor configuration (from/to/hours/offset)
* The time period that the zero tariffs appeared in

**Reply by oymyakon** [reply]
Thanks very much.
Paul.
